app/models/identity.rb

```diff
@@ -0,0 +1,9 @@
+class Identity < ActiveRecord::Base
+  belongs_to :user
+  validates_presence_of :uid, :provider
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.